### PR TITLE
Fix failing tests and update test dependency

### PR DESCRIPTION
**Pull request by AndrewSpeed**
Updates the `hubot-test-helper` dependency to the latest version, fixes the tests which were failing (with minor refactor on setup for applications and environments) and adds a `package-lock.json`.

Test setup has been refactored to remove dependence on other commands, meaning that if a test fails, it's more likely to be a problem with the command under test, and not a command used for setup.

### CODE REVIEW

**Comment by AndrewSpeed** [comment]
CI fails due to this project relying on an older version of `node`, namely `v0.10.36`. Should this be updated to be running against a more up to date release?

**Comment by AlanFoster** [comment]
@AndrewSpeed The version we're currently using on [shopbot](https://github.com/shopkeep/shopbot/blob/master/package.json#L9) is actually `1.0.4`, whilst the latest master sits at `1.1.1`

```json
"dependencies": {
    "hubot": "^2.11.0",
    "hubot-atc": "1.0.4",
```

We should probably decide if we want shopbot to have the lock expiry mechanism or not. In recent years we haven't needed it - so I would be good to nuke it entirely.

To answer your specific question, shopbot is now on node `6.11.1` which (supports promises](http://node.green/#ES2015-built-ins-Promise) - so bumping hubot-atc's to get tests passing LGTM :+1:

File: package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
   },
   "devDependencies": {
     "chai": "*",
-    "hubot-test-helper": "akatz/hubot-test-helper",
+    "hubot-test-helper": "^1.6.0",
     "mocha": "*",
```
Comment (AlanFoster) [inline package.json]: Could we be good citizens and also lock these down to the specific versions that work? :+1:

File: test/atc_test.coffee [hunk 2]
```diff
@@ -6,202 +6,223 @@ chai.use require 'sinon-chai'
...
+
+    room.robot.brain.data.locks = locks
+
   describe "applications", ->
     describe "adding", ->
       context "when there are none", ->
         it 'adds the application', ->
-          room.user.say "duncan", "hubot application add foo"
-          expect(room.robot.brain.data.applications).to.include "foo"
+          @room.user.say("duncan", "hubot application add foo").then =>
```
Comment (AlanFoster) [inline test/atc_test.coffee]: TIL - CoffeeScript is nice here, as it implicitly returns the Promise for mocha to wait on :+1:

File: package.json (hunk 1 above)
Comment (AndrewSpeed) [reply package.json]: :+1: will do

**Comment by AndrewSpeed** [comment]
Thanks for the feedback @AlanFoster I'll update this PR with the changes for the version of node we're using on CI, lock our dependencies to specific versions and then submit a separate PR to remove the lock expiry mechanism

**Comment by AndrewSpeed** [comment]
@AlanFoster before I merge this, just want to double check that you're happy with the updated PR?

**Comment by AlanFoster** [comment]
LGTM :+1:

**Pull request merged**